src/components/checkbox/index.tsx

```diff
@@ -172,10 +196,19 @@ class Checkbox extends Component<CheckboxProps, CheckboxState> {
   }
 
   onPress = () => {
-    const {disabled} = this.props;
-
+    const {disabled, value, required, onValueChange, onChangeValidity} = this.props;
+    
     if (!disabled) {
-      this.props.onValueChange?.(!this.props.value);
+      const newValue = !value;
+
+      onValueChange?.(newValue);
```

Comment:
I feel like this should be called together with `onChangeValidity` after the state was changed, I realize this would make the code a little more awkward since you'll have to add an `else` with another `onValueChange`, so I am conflicted, WDYT?

Reply:
onValueChange will be called either way, the delay doesn't seem much of an issue